### PR TITLE
chore (cleanup): infer TopSite editMode from onEdit prop

### PR DESCRIPTION
**Pull request by rlr**
r? @Mardak

### CODE REVIEW

File: system-addon/content-src/components/TopSites/TopSite.jsx [hunk 1]
```diff
@@ -153,7 +153,7 @@ class TopSite extends React.PureComponent {
               visible={isContextMenuOpen} />
           </div>
         }
-        {props.editMode &&
+        {props.onEdit &&
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSite.jsx]: Curious... does this work or break anything if we do
```js
{!props.onEdit ?
  <div>…
  :
  <div className="edit-menu">…
}
```

Eh.. I suppose the indenting might get wonky ;)

Comment (rlr) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: hmmm. This passes linting:

```
        {!props.onEdit ?
          <div>
          ...
          </div> :
          <div className="edit-menu">
          ...
          </div>
        }
```

But it looks a little funny and the separation is less obvious. I'll leave it for now.

Comment (Mardak) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: Yeah, that's what I was afraid of. A small `:` at the end of the line.

Comment (Mardak) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: Although... oh well too late :p
```js
{!props.onEdit ? (
  <div>
    …
  </div>
) : (
  <div className="edit-menu"
   …
  </div>
)}
```

Eh.. still meh without changing a lot of indenting ;)

**Comment by as-pine-proxy** [comment]
[Treeherder: c0698ec200e3a6788208027d9c0fdc6aabdcd609](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=c0698ec200e3a6788208027d9c0fdc6aabdcd609)

**Pull request merged**